Add a new closure to THM (#25085)
idaholab/moose: modules/thermal_hydraulics/doc/content/source/closures/Closures1PhaseTHM.md

# Closures1PhaseTHM

> !syntax description /Closures/Closures1PhaseSimple

Reviewer: update closure name

The closures added are:

- a wall friction factor model using [ADWallFrictionFunctionMaterial.md]

> - a wall heat transfer coefficient as a [ADWeightedAverageMaterial.md] of the heat transfer coefficients

Reviewer: ```suggestion
- a wall heat transfer coefficient for heat heat transfer
```

Author: done

  defined by the flow channel (deriving from [FlowChannel1Phase.md]) weighted by the heated perimeters of each section.

> The user can choose between a range of correlations for the heat transfer coefficient and friction factor that covers from pipes to rod-bbundles. The following correlatios for the heat transfer coefficient are avaliable:

Reviewer: ```suggestion
The user can choose between a range of correlations for the heat transfer coefficient and friction factor that covers  pipes to rod-bundles. The following correlations for the heat transfer coefficient are available:
```

Author: done

| Gnielinski     | Liquid/Gases  | Pipes        | [ADWallHeatTransferCoefficientGnielinskiMaterial.md]        |
| Wolf-McCarthy  | Gases         | Pipes        | [ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md]      |
| Lyon           | Liquid Sodium | Pipes        | [ADWallHeatTransferCoefficientLyonMaterial.md]              |
| Weisman        | Water         | Rod-bundles  | [ADWallHeatTransferCoefficientWeismanMaterial.md]           |
| Kazimi-Carelli | Liquid Sodium | Rod-bundles  | [ADWallHeatTransferCoefficientKazimiMaterial.md]            |
| Mikityuk       | Liquid Sodium | Rod-bundles  | [ADWallHeatTransferCoefficientMikityukMaterial.md]          |
| Schad          | Liquid Sodium | Rod-bundles  | [ADWallHeatTransferCoefficientSchadMaterial.md]             |

and for the fricition factor we have:

!table id=FF-correlations caption=Avaliable correlations for the friction factor.
| Name          | Geometry    | Reference                            |
|---------------|-------------|--------------------------------------|
| Churchill     | Pipes       | [ADWallFrictionChurchillMaterial.md] |
| Cheng-Todreas | Rod-bundles | [ADWallFrictionChengMaterial.md]     |

> 

Author: done

Additionally, this object defines:

> - a wall temperature material, to be able to retrieve the wall temperature as a material property, using either [ADAverageWallTemperature3EqnMaterial.md], AD[CoupledVariableValueMaterial.md] or [ADTemperatureWall3EqnMaterial.md] depending on the temperature mode of the flow channel (deriving from [FlowChannel1Phase.md]).

Reviewer: ```suggestion
- a wall temperature material, to be able to retrieve the wall temperature as a material property for each heat transfer
```

Author: done
